cmds/rm/rm.go

```diff
@@ -11,81 +11,142 @@
 //     -i: interactive mode
 //     -v: verbose mode
 //     -R: remove file hierarchies
-//     -r: equivalent to -R
 package main
 
 import (
 	"bufio"
 	"flag"
 	"fmt"
+	"io/ioutil"
 	"os"
-	"path"
+	"path/filepath"
+	"syscall"
+	"strings"
+	"log"
 )
 
-var (
-	flags struct {
-		r bool
-		v bool
-		i bool
-	}
-	cmd = "rm [-Rrvi] file..."
-)
+// You can add more flags to this struct.
+type rmFlags struct {
+	recursive   bool
+	verbose     bool
+	interactive bool
+}
 
-func init() {
-	defUsage := flag.Usage
-	flag.Usage = func() {
-		os.Args[0] = cmd
-		defUsage()
+// Helper method for interactive flag
+func iFlag(input *bufio.Scanner, printString string) bool {
+	for{	
+		//replace Println with Printf
+		fmt.Printf("%s (y/n)\n", printString)
+		input.Scan()
+		if input.Err()!= nil{
+			log.Fatalf("Failed at error: %v", input.Err())		
+		}
+		if(strings.Compare(input.Text(), "y") == 0){	
+			return true
+		}
+		if(strings.Compare(input.Text(), "n") == 0){
+			return false
+		}
+		fmt.Printf("Please only enter y or n. You entered %s. \n", input.Text())	
 	}
-	flag.BoolVar(&flags.i, "i", false, "Interactive mode.")
-	flag.BoolVar(&flags.v, "v", false, "Verbose mode.")
-	flag.BoolVar(&flags.r, "R", false, "Remove file hierarchies")
-	flag.BoolVar(&flags.r, "r", false, "Equivalent to -R.")
-	flag.Parse()
 }
 
-func rm(files []string) error {
-	f := os.Remove
-	if flags.r {
-		f = os.RemoveAll
-	}
 
-	workingPath, err := os.Getwd()
+func fileOnly (){
+}
+
+
+func recursiveDelete(file string, flags rmFlags) error {
+	input := bufio.NewScanner(os.Stdin)
+	statval, err := os.Stat(file)
 	if err != nil {
 		return err
 	}
-
-	input := bufio.NewScanner(os.Stdin)
-	for _, file := range files {
-		if flags.i {
-			fmt.Printf("rm: remove '%v'? ", file)
-			input.Scan()
-			if input.Text()[0] != 'y' {
-				continue
+	if statval.IsDir(){
+		// Throws error if flag is not recursive.
+		if !flags.recursive {
+			return &os.PathError{Op: "\nrm:", Path: file, Err: syscall.EISDIR}
```

Comment:
personal opinion: I don't see the need to return anything this complex. What's wrong with just returning
rm: cannot remove ‘%s’: Is a directory with the path?